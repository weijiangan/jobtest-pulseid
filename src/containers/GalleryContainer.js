import { connect } from "react-redux";
import {
  initGallery,
  fetchPhotos,
  selectMode,
  setAllOrder,
  selectSelectedGallery,
  getQuery
} from "../redux/modules/gallery";
import AuthScreen from "../components/Gallery";

function mapStateToProps(state) {
  return {
    selectedGallery: selectSelectedGallery(state),
    query: getQuery(state)
  };
}

export default connect(
  mapStateToProps,
  { initGallery, fetchPhotos }
)(AuthScreen);
