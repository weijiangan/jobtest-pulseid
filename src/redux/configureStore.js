import thunkMiddleware from "redux-thunk";
import { createStore, applyMiddleware, compose, combineReducers } from "redux";
import gallery from "./modules/gallery";

export const reducer = combineReducers({
  gallery
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(initialState) {
  return createStore(
    reducer,
    initialState,
    composeEnhancers(applyMiddleware(thunkMiddleware))
  );
}
