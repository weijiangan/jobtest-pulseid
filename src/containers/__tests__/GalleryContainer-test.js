import React from "react";
import { createStore, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";
import { Provider } from "react-redux";
import { render, fireEvent } from "@testing-library/react";
import { reducer } from "../../redux/configureStore";
import GalleryContainer from "../GalleryContainer";

function renderWithRedux(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunkMiddleware))
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store
  };
}

test("can render with redux with defaults", () => {
  const { getByPlaceholderText, getByText } = renderWithRedux(
    <GalleryContainer />
  );
  getByPlaceholderText("Search images...");
  getByText("Sort by");
  getByText("All");
  getByText("People");
  getByText("Skyline");
  getByText("Travel");
});

test("matches snapshot", () => {
  const { asFragment } = renderWithRedux(<GalleryContainer />);
  expect(asFragment()).toMatchSnapshot();
});

test("hide sort option when not browsing All section", () => {
  const { getByText, queryByText } = renderWithRedux(<GalleryContainer />);
  fireEvent.click(getByText("Animals"));
  expect(queryByText("Sort by")).toBeNull();
  fireEvent.click(getByText("All"));
  expect(queryByText("Sort by")).not.toBeNull();
});

test("test search box", () => {
  const { getByText, getByPlaceholderText } = renderWithRedux(
    <GalleryContainer />
  );
  const input = getByPlaceholderText("Search images...");
  fireEvent.change(input, { target: { value: "travel" } });
  expect(input.value).toBe("travel");
});
