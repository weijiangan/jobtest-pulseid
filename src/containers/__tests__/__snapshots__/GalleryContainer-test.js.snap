// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="topBar"
    >
      <div
        class="container"
      >
        <input
          class="searchBox"
          placeholder="Search images..."
          type="text"
          value=""
        />
        <div
          class="flexBaseline"
        >
          <div
            class="navWrapper"
          >
            <ul
              class="nav"
              data-testid="entries"
            >
              <li
                class="active"
              >
                <button
                  type="button"
                >
                  All
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  People
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Nature
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Skyline
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Abstract
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Animals
                </button>
              </li>
              <li>
                <button
                  type="button"
                >
                  Travel
                </button>
              </li>
            </ul>
          </div>
          <div
            class="flexFixed"
          >
            <span
              class="mr1"
            >
              Sort by 
            </span>
            <select>
              <option
                value="latest"
              >
                latest
              </option>
              <option
                value="oldest"
              >
                oldest
              </option>
              <option
                value="popular"
              >
                popular
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
      style="margin-top: 7.15rem;"
    >
      <div
        class="row"
        data-testid="grid"
      />
    </div>
  </div>
</DocumentFragment>
`;
